src/pygeoprocessing/geoprocessing.py

```diff
@@ -1357,6 +1357,44 @@ def interpolate_points(
         band.WriteArray(raster_out_array, offset['xoff'], offset['yoff'])
 
 
+def align_bbox(geotransform, bbox):
+    """Pad a bounding box so that it aligns with the grid of a given geotransform.
+
+    Ignores row and column rotation.
+
+    Args:
+        geotransform (list): GDAL raster geotransform to align to
+        bbox (list): bounding box in the form [minx, miny, maxx, maxy]
+
+    Returns:
+        padded bounding box in the form [minx, miny, maxx, maxy]
+    """
+    # NOTE: x_origin and y_origin do not necessarily equal minx and miny.
+    # If pixel_width is positive, x_origin = minx
+    # If pixel_width is negative, x_origin = maxx
+    # If pixel height is positive, y_origin = miny
+    # If pixel height is negative, y_origin = maxy
+    x_origin, pixel_width, _, y_origin, _, pixel_height = geotransform
+    if pixel_width == 0 or pixel_height == 0:
+        raise ValueError('Pixel width and height must not be 0')
+    if bbox[2] < bbox[0] or bbox[3] < bbox[1]:
+        raise ValueError('bbox must be in order [minX, minY, maxX, maxY]')
+    if ((pixel_width > 0 and bbox[0] < x_origin) or
+        (pixel_height > 0 and bbox[1] < y_origin) or
+        (pixel_width < 0 and bbox[0] > x_origin) or
+        (pixel_height < 0 and bbox[1] > y_origin)):
+        raise ValueError('bbox must fall within geotransform grid')
+    return [
+        x_origin + abs(pixel_width) * math.floor(
+            (bbox[0] - x_origin) / pixel_width * numpy.sign(pixel_width)),
+        y_origin + abs(pixel_height) * math.floor(
+            (bbox[1] - y_origin) / pixel_height * numpy.sign(pixel_height)),
+        x_origin + abs(pixel_width) * math.ceil(
+            (bbox[2] - x_origin) / pixel_width * numpy.sign(pixel_width)),
```

Comment:
It took me a long time to figure out these equations and I'm still not sure I understand them intuitively. Happy to discuss

Reply:
This would be interesting as a tough interview question

Reply:
I agree, this adjustment would be a neat puzzle-style interview question!  Relevant to the domain, useful and also a little bit obscure.